### ISSUE TITLE
Any planes to support 2019 and 2020 ? 

### ISSUE THREAD

**Issue by kairos0ne** [jonasschmidt/ex_json_schema#73]
Was wondering if you had any plans to support the 2020 schema ?

**Comment by jonasschmidt** [comment]
Yes, absolutely. The main hurdle could be the changed behavior around references. Some things have been simplified/clarified, so it might not be that bad, but I haven't had a chance to look at it in depth yet. I will try to find some time this week and come up with an initial effort estimation.

**Comment by jonasschmidt** [comment]
Ok, some update about the progress. When looking into the new features in the 2019/2020 drafts I realized that there is no point in working on those before properly adding support for a class of references (location-independent identifiers and embedded schemas) that I had dismissed as an edge case in the initial draft 4 implementation. After updating to the latest `master` of the [official test suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite), I found that there are a lot of helpful new tests around those features, and also there's valuable documentation that explains the expected behavior.

I'm now at a point where references are actually working 100% in drafts 4, 6 and 7. From there on it should be much easier to support the later drafts since they are not fundamentally changing that behavior (as far as I can see).

So the TLDR would be: I'm working on it and making some progress, though it could still take a bit longer. I will give updates once I have more information.

**Comment by xinz** [comment]
Hello, could you like update this feature status when possible? I occur a use case need to process the embedded schema from `$ref`, maybe I can help something base on your thoughts. 

Thanks for this open project.

**Comment by jonasschmidt** [comment]
Hey, I'm afraid I haven't made much progress since February, no time to work on it. There's a [branch](https://github.com/jonasschmidt/ex_json_schema/tree/draft-2020-12) with my work so far if you want to have a look, but it might not be in a state where it's easy to pick up. I hope to have more time after easter as soon as I get a bigger project at work out of the way.

**Comment by xinz** [comment]
Got it, thanks.

**Comment by dmolnar99** [comment]
Hey! Are you planning to support 2020 anytime soon?
